packages/interface-connection-manager/src/index.ts

```diff
@@ -1,8 +1,80 @@
 import type { AbortOptions } from '@libp2p/interfaces'
 import type { EventEmitter } from '@libp2p/interfaces/events'
-import type { Connection, MultiaddrConnection } from '@libp2p/interface-connection'
+import type { Connection, ConnectionGater, MultiaddrConnection } from '@libp2p/interface-connection'
 import type { PeerId } from '@libp2p/interface-peer-id'
-import type { Multiaddr } from '@multiformats/multiaddr'
+import type { Multiaddr, Resolver } from '@multiformats/multiaddr'
+import type { Metrics } from '@libp2p/interface-metrics'
+import type { AddressSorter, PeerStore } from '@libp2p/interface-peer-store'
+import type { TransportManager, Upgrader } from '@libp2p/interface-transport'
+export interface ConnectionManagerComponents {
+  peerId: PeerId
+  metrics?: Metrics
+  upgrader: Upgrader
+  peerStore: PeerStore
+  dialer: Dialer
+}
+
+export interface ConnectionManagerConfig {
```

Comment:
We should remove all config interfaces from these modules, each implementation should be free to define its own config, otherwise all implementations have to use the same config which constrains what each implementation can do.